tests/openfhe/end_to_end/dot_product_8_test.cpp

```diff
@@ -21,18 +21,25 @@ namespace openfhe {
 TEST(DotProduct8Test, RunTest) {
   // TODO(#661): Generate a helper function to set up CryptoContext based on
   // what is used in the generated code.
-  CCParams<CryptoContextBGVRNS> parameters;
-  parameters.SetMultiplicativeDepth(2);
-  parameters.SetPlaintextModulus(65537);
-  CryptoContext<DCRTPoly> cryptoContext = GenCryptoContext(parameters);
-  cryptoContext->Enable(PKE);
-  cryptoContext->Enable(KEYSWITCH);
-  cryptoContext->Enable(LEVELEDSHE);
+  // CCParams<CryptoContextBGVRNS> parameters;
+  // parameters.SetMultiplicativeDepth(2);
+  // parameters.SetPlaintextModulus(65537);
 
-  KeyPair<DCRTPoly> keyPair;
-  keyPair = cryptoContext->KeyGen();
-  cryptoContext->EvalMultKeyGen(keyPair.secretKey);
-  cryptoContext->EvalRotateKeyGen(keyPair.secretKey, {1, 2, 4, 7});
+  // CryptoContext<DCRTPoly> cryptoContext = GenCryptoContext(parameters);
+  // cryptoContext->Enable(PKE);
+  // cryptoContext->Enable(KEYSWITCH);
+  // cryptoContext->Enable(LEVELEDSHE);
+  // KeyPair<DCRTPoly> keyPair = cryptoContext->KeyGen();
+  // PublicKey<DCRTPoly> publicKey = keyPair.publicKey;
+  // PrivateKey<DCRTPoly> secretKey = keyPair.secretKey;
+  // cryptoContext->EvalMultKeyGen(secretKey);
+  // cryptoContext->EvalRotateKeyGen(secretKey, {1, 2, 4, 7});
+  auto cryptoContext = dot_product__generate_crypto_context();
+  auto keyPair = cryptoContext->KeyGen();
+  auto publicKey = keyPair.publicKey;
+  auto secretKey = keyPair.secretKey;
```

Comment:
Should there be a option in the pass to add public/private key gen directly into the configuration helper?

Reply:
I agree that this is a good idea.

There is one minor issue though. When I initially created this PR, I also wanted to model the Keygen functionality in the Openfhe dialect like: (pub_key, sec_key) = Openfhe.keygen(context).

However, when I tried to generate code using this, it failed at [OpenFhePkeEmitter](https://github.com/google/heir/blob/9cad9ed31d826550fdc0ce54068a85ac0167a949/lib/Target/OpenFhePke/OpenFhePkeHeaderEmitter.cpp#L78) because it only supports single return type, causing code generation to fail. As far as I know, it seems like mlir no longer supports tuple types. Is there any better solution?

Anyway, I also think it would be great if we could generate the keygen as a helper function, and it might be good to leave a todo for the same issue :)

Reply:
> However, when I tried to generate code using this, it failed at [OpenFhePkeEmitter](https://github.com/google/heir/blob/9cad9ed31d826550fdc0ce54068a85ac0167a949/lib/Target/OpenFhePke/OpenFhePkeHeaderEmitter.cpp#L78) because it only supports single return type, causing code generation to fail. As far as I know, it seems like mlir no longer supports tuple types.

Oh! I think actually that was a restriction that was legacy from the Verilog emitter. I expect that we don't need that here. See this MLIR [test](https://github.com/google/heir/blob/62d5c959f741e003fdae19b9a558ee2fa474dd50/tests/jaxite/emit_jaxite.mlir#L17) for an example of multiple return values.

Agree, I'll note it in the issue thread as a comment! I think I can help take care of that and fix the code assumptions.